File: QID-3529-SFSLossBern/SFSLossBern.R
# Close all plots and clear variables
graphics.off()
rm(list = ls(all = TRUE))

# parameter settings
h   = 0.001
p   = seq(0, 0.99, h)
m   = 100
k   = seq(0, m, 1)
L1  = matrix(0, m + 1, 1)
L2  = L1
L3  = L1
L4  = L1
L5  = L1
L6  = L1

fp1 = dbeta(p, 5, 25)
fp2 = dbeta(p, 10, 25)
fp3 = dbeta(p, 15, 25)

for (i in 1:(m + 1)) {
    L1[i] = sum(dbinom(k[i], m, p) * fp1 * h)
    L2[i] = sum(dbinom(k[i], m, p) * fp2 * h)
    L3[i] = sum(dbinom(k[i], m, p) * fp3 * h)
}

plot(k, L1, type = "l", col = "blue", lwd = 2, xlab = "", ylab = "")
lines(k, L2, col = "blue", lwd = 2)
lines(k, L3, col = "blue", lwd = 2)

fp4 = dbeta(p, 5, 45)
fp5 = dbeta(p, 10, 90)
fp6 = dbeta(p, 20, 180)

for (i in 1:(m + 1)) {
    L4[i] = sum(dbinom(k[i], m, p) * fp4 * h)
    L5[i] = sum(dbinom(k[i], m, p) * fp5 * h)
    L6[i] = sum(dbinom(k[i], m, p) * fp6 * h)
}
dev.new()
plot(k, L4, type = "l", col = "blue", lwd = 2, xlim = c(0, 30), ylim = c(0, 0.12), 
    xlab = "", ylab = "")
lines(k, L5, col = "blue", lwd = 2)
lines(k, L6, col = "blue", lwd = 2) 

File: QID-3529-SFSLossBern/SFSLossBern.m
%------------------------------------------------------------------------
% Book:         SFS
% ----------------------------------------------------------------------
% Quantlet:     SFSLossBern
% ----------------------------------------------------------------------
% Description:  Loss distribtion in the simplified Bernoulli model.
%               Corresponds to exercise 18.4 in SFS.
%------------------------------------------------------------------------
% Usage:        -
%-----------------------------------------------------------------------
% Inputs:       None
%-----------------------------------------------------------------------
% Output:       Plots of the loss distribution in the simplified Bernouli  
%               model with default probabilities comming from Beta
%               distribution.
% ----------------------------------------------------------------------
% Example:      the example is produced for the 100 obligors and Beta
%               parameters:  (5,25), (10,25), (15,25), (5,45), (10,90), 
%               (20,180) 
%------------------------------------------------------------------------
% Author  :     Szymon Borak 20090731
%------------------------------------------------------------------------

clear
close all
clc

h  = 0.001;
p  = 0:h:0.99;
m  = 100;
k  = 0:m;
L1 = zeros(1,m+1);
L2 = L1;
L3 = L1;


fp1 = betapdf(p,5,25);
fp2 = betapdf(p,10,25);
fp3 = betapdf(p,15,25);


for i=1:(m+1)
    L1(i) = sum(binopdf(k(i),m,p).*fp1*h);
    L2(i) = sum(binopdf(k(i),m,p).*fp2*h);
    L3(i) = sum(binopdf(k(i),m,p).*fp3*h);
end
plot(k,L1,k,L2,k,L3,'Linewidth',2,'Color',[0,0,1])

fp4 = betapdf(p,5,45);
fp5 = betapdf(p,10,90);
fp6 = betapdf(p,20,180);




